File: Programação Funcional/Básicos/Haskell/05.hs
{- 
	Faça um programa que retorne True caso a entrada seja menor que -1 ou (maior que 1 E múltiplo de 2), e False caso contrário.
-}
condicao :: Int -> Bool
condicao x = (x < -1) || (x > 1 && x `mod` 2 == 0)

-- Função principal para testar
main :: IO ()
main = do
    print (condicao (-3))  -- Deve retornar True
    print (condicao 4)   -- Deve retornar True
    print (condicao 5)   -- Deve retornar False

File: Programação Funcional/Funções/Scala/Funções Simples/05.scala
object Main {
  // Função que retorna true se a entrada for menor que -1 ou (maior que 1 e múltiplo de 2)
  def condicao(x: Int): Boolean = (x < -1) || (x > 1 && x % 2 == 0)

  // Função principal
  def main(args: Array[String]): Unit = {
    println(condicao(-3))  // Deve retornar True
    println(condicao(4))   // Deve retornar True
    println(condicao(5))   // Deve retornar False
  }
}
